### PR TITLE
Implement Mailchimp signup

### LINKED ISSUE #283: Home: Newsletter sign up
Entire users to sign up for the OCW newsletter

#### Designs and Mockups

from https://invis.io/93Y7PJ6REFN#/426576569_OCW_NexGen_Homepage_07-27-2020_Updates_-Shareable_Prototype-

image

#### Acceptance Criteria:

- [ ] Use a mailchimp form, based on the one used at https://ocw.mit.edu/subscribe/ -- if we needs modification in the mailchimp account, let me know
- [ ] if possible return a success message on the same page instead of loading a new page

### PR DESCRIPTION
**Pull request by noisecapella**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Fixes #283 

#### What's this PR do?
Implements mailchimp signup. We need to add `MAILCHIMP_USER_ID` and `MAILCHIMP_AUDIENCE_ID` environment variables to make this work and have them be present when hugo is run.

#### How should this be manually tested?
- Create a mailchimp account and get the user and audience ID. You can find these by going to the mailchimp dashboard and then clicking on Audience => Signup Forms => Embedded Forms => Condensed, for example. In the `action` attribute of `form` there is a url. The `u=` part is the user id and the `id=` is the audience id.
 - Set these values to `MAILCHIMP_USER_ID` and `MAILCHIMP_AUDIENCE_ID` in `.env`
 - Run `docker-compose up`
 - Fill out the email signup form at the bottom. You should see a success message or some variety of failure message.
 - You should be able to both press enter to submit and click the 'Sign me up' button.

#### Other context
This uses an undocumented feature I found on stackoverflow. I can't find anything about `post-json` on mailchimp's site. Hopefully it's reliable? https://stackoverflow.com/questions/8425701/ajax-mailchimp-signup-form-integration

#### Screenshots (if appropriate)
The empty form: Screenshot from 2020-11-04 16-24-51

After filling out and submitting the form, you should see this:
Screenshot from 2020-11-04 16-13-44

If you're already subscribed:
Screenshot from 2020-11-04 16-14-12

If you didn't type in a valid email:
Screenshot from 2020-11-04 16-16-01

These messages come from Mailchimp's response.

### CODE REVIEW

File: src/js/mailchimp.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    const response = await $.ajax({
+      url:      url,
+      data:     $form.serialize(),
+      dataType: "jsonp",
+      method:   $form.attr("method")
+    })
+    const { result, msg } = response
+    if (result === "error") {
+      $form.html(msg)
```
Comment (mbertrand) [inline src/js/mailchimp.js:26]: This replaces the input with the error message, preventing the user from correcting a mistake.  Can it be appended underneath the input instead?

Screen Shot 2020-11-05 at 3 43 02 PM

**Review by mbertrand — CHANGES_REQUESTED** [review]

**Comment by noisecapella** [comment]
How is this?
Screenshot from 2020-11-05 17-33-24

Screenshot from 2020-11-05 17-33-46

**Review by mbertrand — COMMENTED** [review]
A couple minor things:

- If I enter an invalid email with no '@', the error message from mailchimp starts with a "0 - ".  This error message is coming from mailchimp so I don't know if it's worth stripping that out or not since it may change in the future.
- The entire submit button appears to be clickable, but it is only submitted if I click directly over the "SIGN ME UP" text.

**Review by mbertrand — APPROVED** [review]
👍

**Comment by noisecapella** [comment]
@mbertrand per our meeting I'm going to convert the jQuery code to pure javascript before merging

**Comment by noisecapella** [comment]
@mbertrand on second thought I'm going to leave it in for now. Someone can fix this in #316. It seems nontrivial to replace the cross domain ajax hack that jquery uses, though there's probably a library somewhere to do it.

**Pull request merged**